tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -584,6 +584,39 @@ final class TestingServer(
     }
   }
 
+  private def rangeFromString[T](
+      filename: String,
+      original: String,
+      root: AbsolutePath,
+      replaceWith: String = ""
+  )(
+      fn: (String, TextDocumentIdentifier, l.Range) => T
+  ): Future[T] = {
+    val startOffset = original.indexOf("<<")
+    val endOffset = original.indexOf(">>")
+    if (startOffset < 0) sys.error(s"missing <<\n$original")
+    if (endOffset < 0) sys.error(s"missing >>\n$original")
+    if (startOffset > endOffset)
+      sys.error(s"invalid range, >> must come after <<\n$original")
+    val text =
+      original
+        .replaceAllLiterally("<<", replaceWith)
+        .replaceAllLiterally(">>", replaceWith)
+    val input = m.Input.String(text)
+    val path = root.resolve(filename)
+    path.touch()
+    val pos = m.Position.Range(input, startOffset, endOffset - "<<>>".length())
```

Comment:
the `endOffset` is adjusted to account for the removal of the range markers

Reply:
A lot of this code seem duplicate to what we already have. Would it be possible to reuse the methods that we use for changing to the position in params?

Reply:
@tgodzik which method? Anyway, sure, I'll open a tech debt issue.